src/Redux/Rockets/RocketsSlice.js

```diff
@@ -0,0 +1,54 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+const GET = 'space-travelers-hub/rocketsReducer/GET';
+const BOOK = 'space-travelers-hub/rocketsReducer/BOOK';
+
+// Book rocket
+const reserveRocket = (state, payload) => {
+  const newstate = state.map((rocket) => {
+    if (rocket.id !== payload) return rocket;
+    return { ...rocket, reserved: !rocket.reserved };
+  });
+  return newstate;
+};
+
+//  Reducer
+const rocketsReducer = (state = [], action) => {
+  switch (action.type) {
+    case `${GET}/fulfilled`:
+      return Object.keys(action.payload).map((key) => ({
+        id: action.payload[key].id,
+        rocket_name: action.payload[key].rocket_name,
+        description: action.payload[key].description,
+        flickr_images: action.payload[key].flickr_images[0],
+        reserved: false,
+      }));
```

Comment:
- [ ] **[OPTIONAL]:** I would suggest that you keep your reducers as clean as possible, this is because they are intended to be pure functions that just receive an action and modify the state accordingly, having complex logic, like organizing the data from the API, would be better handled by the thunk you created to fetch it! I strongly suggest you implement this. 🤓 🔧